### PR TITLE
[cmake] Add and export Modules_CUDA_fix

### PR DESCRIPTION
**Pull request by Yangqing**
This PR puts the discretional inclusion of Modules_CUDA_fix to public/cuda.cmake and also installs the corresponding files. Manually verified the build process but the installation path is not tested (only did a dummy install and eyeballed that the files are there).

### CODE REVIEW

**Review by soumith — APPROVED** [review]
fixed worked locally on my ubuntu 16.04 install which was broken on master

**Comment by Yangqing** [comment]
cc @orionr FYI - no action items needed.

File: cmake/public/cuda.cmake [hunk 1]
```diff
@@ -1,5 +1,13 @@
 # ---[ cuda
 
+# Check if we need to use the new FindCUDA functionalities, which are included
+# since 3.7.0. Also, we know that VS2017 needs the new FindCUDA functionality,
+# so we will simply enable it for the whole Windows build.
+if (MSVC OR ${CMAKE_VERSION} VERSION_LESS 3.7.0)
```
Comment (ezyang) [inline cmake/public/cuda.cmake]: This is wrong, we are tracking cmake MASTER; 3.7.0 is the wrong commit.

Comment (ezyang) [reply cmake/public/cuda.cmake]: All of the modifications in `Modules_CUDA_fix` exist in cmake master. So, hypothetically, once they do a release we could pin to that.

Though, I do agree it's less error prone if we just always include our own copy.

**Pull request merged**